src/main/scala/org/neo4j/spark/reader/Neo4jDataSourceReader.scala

```diff
@@ -27,7 +31,14 @@ class Neo4jDataSourceReader(private val options: DataSourceOptions, private val
   override def planInputPartitions: util.ArrayList[InputPartition[InternalRow]] = {
     val schema = readSchema()
     val factoryList = new java.util.ArrayList[InputPartition[InternalRow]]
-    factoryList.add(new Neo4jInputPartitionReader(neo4jOptions, schema, jobId))
+    factoryList.add(new Neo4jInputPartitionReader(neo4jOptions, filters, schema, jobId))
     factoryList
   }
+
+  override def pushFilters(filtersArray: Array[Filter]): Array[Filter] = {
+    filters = filtersArray
```

Comment:
scusa ma invecere che mettere l'if dentro i vari pezzi non possiamo fare così:
```suggestion
    if (neo4jOptions.pushdownFiltersEnabled) {
        filters = filtersArray
    }
```

così lo gestiamo in un solo punto

Reply:
no, altrimenti non applica filtri nemmeno su Spark 😞